gadget/gadget.go

```diff
@@ -375,6 +371,24 @@ func ReadInfo(gadgetSnapRootDir string, constraints *ModelConstraints) (*Info, e
 	return InfoFromGadgetYaml(gmeta, constraints)
 }
 
+// ReadInfo reads the gadget specific metadata from meta/gadget.yaml in the snap
+// root directory. If constraints is nil, ReadInfo will just check for
+// self-consistency, otherwise rules for the classic or system seed cases are
+// enforced.
+func ReadInfo(gadgetSnapRootDir string, constraints *ModelConstraints) (*Info, error) {
+	gadgetYamlFn := filepath.Join(gadgetSnapRootDir, "meta", "gadget.yaml")
+	return readInfo(ioutil.ReadFile, gadgetYamlFn, constraints)
+}
+
+// ReadInfoFromSnapFile reads the gadget specific metadata from
+// meta/gadget.yaml in the given snapf container. If constraints is
```

Comment:
I would s/snapf/snap/, I know is the argument name, but it just read too much like a typo here (for me)